src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -178,15 +211,26 @@ def _parse_frontend_config(config):
     >>> _parse_frontend_config('civetweb port=[::1]:8443s')
     (8443, True)
 
+    >>> _parse_frontend_config('civetweb port=127.0.0.1:8443s+8000')
+    (8443, True)
+
+    >>> _parse_frontend_config('civetweb port=127.0.0.1:8080+443s')
+    (8080, False)
+
     >>> _parse_frontend_config('civetweb port=xyz')
     Traceback (most recent call last):
     ...
-    LookupError: Failed to determine RGW port
+    LookupError: Failed to determine RGW port from "civetweb port=xyz"
 
     >>> _parse_frontend_config('civetweb')
     Traceback (most recent call last):
     ...
-    LookupError: Failed to determine RGW port
+    LookupError: Failed to determine RGW port from "civetweb"
+
+    >>> _parse_frontend_config('mongoose port=8080')
+    Traceback (most recent call last):
+    ...
+    LookupError: Failed to determine RGW port from "mongoose port=8080"
```

Comment:
Taking into account the amount of doc tests, would it be better to have these as unit tests in a separate test file ?

Reply:
IMO using doctest is enough here because it is also a simple documentation that shows the developer what syntax is supported by the function. A unit test can't do that because it's located in a separate place.

Reply:
I see what you mean, my point here is doctests are cluttering the source code file up and IMHO it's better to follow Clean Code guidelines and keeping files short:
http://www.itamarweiss.com/code/2017/11/05/Clean-code-main-takeaways.html
Although unit tests are located in a separate file, they can contain comments/clarifications if needed.

Reply:
Tests have been relocated into separate file.